### PR TITLE
ISSUE-104: kafka examples code should run against secure clusters.

### PR DESCRIPTION
**Pull request by harshach**

### CODE REVIEW

File: docs/examples.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Examples
+============
+
+SchemaRegistry comes with examples of integration into Kafka. You can find the code
+here https://github.com/hortonworks/registry/tree/master/examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples
```
Comment (satishd) [inline docs/examples.rst]: You may want to give that URL as hyperlink for "here"

File: docs/examples.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+bootstrap.servers=kafka_host1:6667,kafka_host2:6667
+
+schema.registry.url=http://regisry_host:9090/api/v1
+
+
+5. java -jar avro-examples-0.2.jar -d data/truck_events.csv -p data/kafka-producer.props -sm -s data/truck_events.avsc
+6. The above command will register truck_events schema in data/truck_events.avsc into registry and ingests 200 messages into topic “truck_events_stream”
+
+
```
Comment (satishd) [inline docs/examples.rst]: May need to add section for running consumers with deserializer also.

File: docs/examples.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+1. Issue ACLs on the topic you are trying to ingest
+
+2. create kafka topic
+   bin/kafka-acls.sh --authorizer kafka.security.auth.SimpleAclAuthorizer --authorizer-properties zookeeper.connect=pm-eng1-cluster2.field.hortonworks.com:2181 --add
+       --allow-principal User:principal_name --allow-host "*" --operation All --topic truck_events_stream
+
+   2.1 make sure you replace "principal_name" with the username you are trying to ingest
+
+3.  edit data/kafka-producer.props , add security.protocol=SASL_PLAINTEXT
```
Comment (satishd) [inline docs/examples.rst]: May want to give respective configuration for kafka consumer.

File: examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples/avro/KafkaAvroSerDesApp.java [hunk 4]
```diff
@@ -118,17 +117,6 @@ public void sendMessages(String payloadJsonFile) throws Exception {
         }
     }
 
-    private Map<String, Object> createProducerConfig(Properties props) {
-        String bootstrapServers = props.getProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG);
-        Map<String, Object> config = new HashMap<>();
-        config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
-        config.putAll(Collections.singletonMap(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name(), props.get(SCHEMA_REGISTRY_URL)));
-        config.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
-        config.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class.getName());
```
Comment (satishd) [inline examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples/avro/KafkaAvroSerDesApp.java]: key/value serializer should be updated in kafka-producer.props.

File: examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples/avro/KafkaAvroSerDesApp.java [hunk 5]
```diff
@@ -166,22 +152,6 @@ public void consumeMessages() throws Exception {
         }
     }
 
-    private Map<String, Object> createConsumerConfig(Properties props) {
-        String bootstrapServers = props.getProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG);
-        Map<String, Object> config = new HashMap<>();
-        config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
-        config.putAll(Collections.singletonMap(SchemaRegistryClient.Configuration.SCHEMA_REGISTRY_URL.name(), props.get(SCHEMA_REGISTRY_URL)));
-        config.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
-        config.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, KafkaAvroDeserializer.class.getName());
```
Comment (satishd) [inline examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples/avro/KafkaAvroSerDesApp.java]: key/value deserializer should be updated in kafka-consumer.props.

File: docs/examples.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+4. java -Djava.security.auth.login.config=/etc/kafka/conf/kafka_client_jaas.conf -jar avro-examples-0.3.0-SNAPSHOT.jar -d data/truck_events_json -p data/kafka-producer.props -sm -s data/truck_events.avsc
```
Comment (HeartSaVioR) [inline docs/examples.rst]: Assuming that jaas config file is from Ambari installation, it doesn't contain principal and keytab so need to kinit with proper principal before executing command. I was doing same to test Streamline with secure cluster.

**Comment by satishd** [comment]
@harshach There are couple of changes required in [KafkaAvroSerDesWithKafkaServerTest](https://github.com/hortonworks/registry/blob/master/schema-registry/rest-service/src/test/java/com/hortonworks/registries/schemaregistry/avro/KafkaAvroSerDesWithKafkaServerTest.java) to address build failures because of kafka version upgrade to 0.10.2.1.
- Replace `kafka.common.TopicExistsException` with `org.apache.kafka.common.errors.TopicExistsException`
- Replace `EmbeddedSingleNodeKafkaCluster` with `EmbeddedKafkaCluster`

**Comment by harshach** [comment]
@satishd addressed the above comments. Please review and merge it in.

File: docs/examples.rst (hunk 2 above)
Comment (harshach) [inline docs/examples.rst]: done

File: docs/examples.rst (hunk 3 above)
Comment (harshach) [inline docs/examples.rst]: done

File: examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples/avro/KafkaAvroSerDesApp.java (hunk 4 above)
Comment (harshach) [inline examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples/avro/KafkaAvroSerDesApp.java]: done

File: examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples/avro/KafkaAvroSerDesApp.java (hunk 5 above)
Comment (harshach) [inline examples/schema-registry/avro/src/main/java/com/hortonworks/registries/schemaregistry/examples/avro/KafkaAvroSerDesApp.java]: done

**Pull request merged**